gradle-jdks-setup/src/main/java/com/palantir/gradle/jdks/setup/CaResources.java

```diff
@@ -53,121 +65,144 @@ public CaResources(ILogger logger) {
-                    new ProcessBuilder(importCertificateCommand), this::writeStdOutput, this::writeStdError);
-        } catch (IOException e) {
-            throw new RuntimeException("Unable to import the certificate to the jdk", e);
+            return Collections.list(keyStore.aliases()).stream()
+                    .map(alias -> {
+                        try {
+                            return Optional.ofNullable(keyStore.getCertificate(alias));
+                        } catch (KeyStoreException e) {
+                            throw new RuntimeException("Failed to load keystore", e);
+                        }
+                    })
+                    .flatMap(Optional::stream)
+                    .filter(certificate -> X509Certificate.class.isAssignableFrom(certificate.getClass()))
+                    .map(X509Certificate.class::cast)
+                    .collect(Collectors.toSet());
+        } catch (KeyStoreException e) {
+            throw new RuntimeException(e);
         }
     }
 
-    private void writeStdOutput(InputStream inputStream) {
-        CommandRunner.processStream(inputStream, logger::log);
+    private static boolean isCertUsedForTls(X509Certificate certificate) {
+        return hasCaCertUsage(certificate) && isTlsServerAuthentication(certificate);
     }
 
-    private void writeStdError(InputStream inputStream) {
-        CommandRunner.processStream(inputStream, logger::logError);
+    private static boolean isValid(X509Certificate certificate) {
+        try {
+            certificate.checkValidity();
+            return true;
+        } catch (CertificateExpiredException | CertificateNotYetValidException e) {
+            return false;
+        }
     }
 
-    private static boolean isCertificateInTruststore(Path jdkInstallationDirectory, String alias) {
-        try {
-            String keytoolPath = jdkInstallationDirectory
-                    .resolve("bin/keytool")
-                    .toAbsolutePath()
-                    .toString();
-            List<String> checkCertificateCommand = List.of(
-                    keytoolPath,
-                    "-list",
-                    "-storepass",
-                    "changeit",
-                    "-alias",
-                    alias,
-                    "-keystore",
-                    jdkInstallationDirectory.resolve("lib/security/cacerts").toString());
-            CommandRunner.runWithOutputCollection(new ProcessBuilder().command(checkCertificateCommand));
+    private static boolean hasCaCertUsage(X509Certificate certificate) {
+        boolean[] keyUsage = certificate.getKeyUsage();
+        if (keyUsage == null) {
             return true;
-        } catch (Exception e) {
-            if (e.getMessage().contains(String.format("Alias <%s> does not exist", alias))) {
-                return false;
+        }
+        // digitalSignature and keyEncipherment are enabled
+        if (keyUsage[0] && keyUsage[2]) {
```

Comment:
I'd make the 0 and 2 (and 5) here constants so we can give them a name like `DIGITAL_SIGNATURE` so it looks like `keyUsage[DIGITAL_SIGNATURE]` and is a bit more self documenting.

Reply:
oops, forgot I had suggested this. No worries.